testutils/rpc.go

```diff
@@ -48,3 +50,39 @@ func (td *TrackingDialer) DialFunc(
 	}
 	return conn, cached, err
 }
+
+// ServerTransportStream implements grpc.ServerTransportStream and can be used to test setting
+// metadata in the gRPC response header.
+type ServerTransportStream struct {
+	mu sync.Mutex
+	grpc.ServerTransportStream
+	md metadata.MD
+}
+
+// NewServerTransportStream creates a new ServerTransportStream.
+func NewServerTransportStream() *ServerTransportStream {
+	return &ServerTransportStream{}
+}
+
+// SetHeader implements grpc.ServerTransportStream.
```

Comment:
[nit] might be worth explaining how this is supposed to differ from SendHeader and why in this mock up the code is the same

Reply:
Good catch